brainvisa/toolboxes/spm/processes/tools/SPMtissuesvolume.py

```diff
@@ -120,183 +127,105 @@ def initialization(self):
-            csf_arr[mask_arr == 0] = False
-            tiv_arr[mask_arr == 0] = False
-            #Calculation of the volumes
-            wm_vol = wm_arr.sum()*vox_vol/1000.
-            gm_vol = gm_arr.sum()*vox_vol/1000.
-            csf_vol = csf_arr.sum()*vox_vol/1000.
-            tiv = tiv_arr.sum()*vox_vol/1000.
+        tiv = compute_vol(tiv)
+        wm_vol = compute_vol(white)
+        gm_vol = compute_vol(grey)
+        csf_vol = compute_vol(csf)
 
-        #Write in the output file
-        f.write(subject+","+
-                str(round(gm_vol,3))+","+
-                str(round(wm_vol,3))+","+
-                str(round(csf_vol,3))+","+
-                str(round(tiv,3))+"\n")
+    elif self.mode == 'proba_max':
+        tmp_w = context.temporary('gz compressed NIFTI-1 image')
+        context.runProcess('computeOneTissueMask',
+                           native_prob_map=self.white_proba_map,
+                           others_prob_maps=[self.grey_proba_map, self.csf_proba_map,
+                                             self.skull_proba_map, self.scalp_proba_map],
+                           native_mask=tmp_w,
+                           resolve_equal_probability=True)
+        tmp_g = context.temporary('gz compressed NIFTI-1 image')
+        context.runProcess('computeOneTissueMask',
+                           native_prob_map=self.grey_proba_map,
+                           others_prob_maps=[self.white_proba_map, self.csf_proba_map,
+                                             self.skull_proba_map, self.scalp_proba_map],
+                           native_mask=tmp_g,
+                           resolve_equal_probability=True)
+        tmp_c = context.temporary('gz compressed NIFTI-1 image')
+        context.runProcess('computeOneTissueMask',
+                           native_prob_map=self.csf_proba_map,
+                           others_prob_maps=[self.grey_proba_map, self.white_proba_map,
+                                             self.skull_proba_map, self.scalp_proba_map],
+                           native_mask=tmp_c,
+                           resolve_equal_probability=True)
+        white = aims.read(tmp_w.fullPath(), border=2)
+        grey = aims.read(tmp_g.fullPath(), border=2)
+        csf = aims.read(tmp_c.fullPath(), border=2)
+        tiv = white + grey + csf
+        aims.AimsConnectedComponent(tiv, aims.Connectivity.CONNECTIVITY_26_XYZ,
+                                    0, True, 0, 0, 1)
+        cltiv = aimsalgo.AimsMorphoClosing(tiv, 1)
+        white[np.where(cltiv.np == 0)] = 0
+        grey[np.where(cltiv.np == 0)] = 0
+        csf[np.where(cltiv.np == 0)] = 0
 
-    context.progress(nb, nb, process=self)
+        tiv = compute_vol(tiv)
+        wm_vol = compute_vol(white)
+        gm_vol = compute_vol(grey)
+        csf_vol = compute_vol(csf)
+
+    #Write in the output file
+    f.write(f'{subject},{str(gm_vol)},{str(wm_vol)},{str(csf_vol)},{str(tiv)}\n')
```

Comment:
What you did works well.
In case you can also use the csv python module : 
```python
import csv

with open(filepath, "w") file:
    file_writer = csv.writer(file)
    # Write header
    file_writer.writerow(["subject", "grey", "white", "csf", "tiv"])
    # Write data
    file_writer.writerow([subject, gm_vol, wm_vol, csf_vol, tiv])
```

Reply:
Good to know, thanks !